flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -781,6 +787,32 @@ class TaskManager(
     }
   }
 
+  private def handleRequestTaskManagerLog
+  (sender: ActorRef, requestType: LogTypeRequest, jobManager: ActorRef):
+  Unit = {
+    if (blobService.isDefined && jobManager.path.address.host.isDefined) {
+      //create new Thread to upload log, to not block the TM for too long
+      new Thread() {
+        override def run(): Unit = {
+          val jmHost = jobManager.path.address.host.get
+          val port = blobService.get.getPort
```

Comment:
Maybe we should extend the `BlobService` to return the `serverAddress` and not only the port.